api/src/wfl/executor.clj

```diff
@@ -521,6 +547,49 @@
          (map link-retry-to-original (sort-by :entity original-records))
          (write-retries (utc-now)))))
 
+;; Visible for testing
+(def retry-invalid-submission-error-message
+  "Missing or invalid Terra Workspace submission ID")
+
+(defn ^:private retry-submission-validation-error
+  "Return error information to throw
+  if `submission` is an invalid retry argument."
+  [submission]
+  (when-not (s/valid? ::submission submission)
+    {:message retry-invalid-submission-error-message}))
+
+;; Visible for testing
+(def retry-unsupported-status-error-message
+  "Retry unsupported for requested workflow status")
+
+(defn ^:private retry-status-validation-error
+  "Return error information to throw
+  if `status` is an invalid retry argument."
+  [status]
+  (when-not (cromwell/retry-status? status)
+    {:message retry-unsupported-status-error-message
+     :data    {:supported-statuses cromwell/retry-status?}}))
+
+(defn ^:private terra-executor-throw-if-invalid-retry-filters
```

Comment:
The compiler-style "throw on the first error you find" approach to validating staged workload requests has made for some frustrating debugging cycles: submit, fix a bug, submit, fix a bug, etc.

Here I gather validation errors for all the parameters I'm checking, and throw at the end if necessary.  Kind of like a An example request whose body would have passed the spec but has two bad filters, and its response:
```
curl -X POST "http://localhost:3000/api/v1/workload/6b1724db-fb1c-49ad-b5fa-89bcf6aa58bd/retry" \
 -H 'Authorization: Bearer '$(gcloud auth print-access-token) \
 -H "Content-Type: application/json"  \
 -d '{ "status": "Running" }'
{
  "message" : "Cannot retry workload: invalid workflow filters.",
  "uri" : "/api/v1/workload/6b1724db-fb1c-49ad-b5fa-89bcf6aa58bd/retry",
  "details" : {
    "workload" : "6b1724db-fb1c-49ad-b5fa-89bcf6aa58bd",
    "filters" : {
      "status" : "Running"
    },
    "validation-errors" : [ "Missing or invalid Terra Workspace submission ID", "Retry unsupported for requested workflow status" ],
    "supported-statuses" : [ "Failed", "Succeeded", "Aborted" ]
  }
}
```

This will also help when we expand retry capability to allow specification of workflow UUIDs.